### PR TITLE
[IMP] *: enforce batch creation support for main Odoo models

### PR DESCRIPTION
**Pull request by Feyensv**
This PR adds post-install tests to enforce Odoo main models always support batch record creation (i.e. that all `create` overrides are `api.model_create_multi`).

* account: main models
* base: `res.partner` & `res.users`
* mail: main models
* product: `product.product` & `product.template`
* purchase: `purchase.order.line`
* sale: `sale.order.line`
* stock: `stock.move.line`, `stock.move`, `stock.inventory.line`

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by seb-odoo** [comment]
Nice idea to have those tests.

Don't we want some kind of test helper to avoid copy/pasting that same patch every time we want to do that assert?

File: addons/stock/tests/test_robustness.py [hunk 1]
```diff
@@ -221,3 +221,9 @@ def test_lot_id_product_id_mix(self):
                 'location_dest_id': move2.location_dest_id.id,
             })]})
 
+
+@tagged('post_install', '-at_install')
+class TestPerformance(TransactionCase):
+    def test_sml_create_batch(self):
+        """Enforce stock.move.line create overrides support batch record creation."""
+        self.assertModelCreateMulti("stock.move.line")
```
Comment (MiquelRForgeFlow) [inline addons/stock/tests/test_robustness.py:229]: I would add `stock.move` and `stock.inventory.line`

**Comment by MiquelRForgeFlow** [comment]
and maybe also `pos.order.line` for point_of_sale

**Comment by seb-odoo** [comment]
If we are going too far with this I feel like it would be time to prevent non-multi create at ORM level?

**Comment by Feyensv** [comment]
Currently, the main point was to avoid regression on models where I specifically fixed it to support batch record creation (mainly for sped up database population).
I'm not sure we want to really enforce it everywhere/change the default API atm ... (even more seeing some reverted API changes done recently).

**Comment by seb-odoo** [comment]
Is this PR still relevant?

**Comment by Feyensv** [comment]
Indeed, not relevant anymore, create multi is enforced since #80824

**Comment by seb-odoo** [comment]
> Indeed, not relevant anymore, create multi is enforced since #80824

Do you know why we still have the decorator in that PR if multi is supposedly now a necessity? Can't it just be like that by default?

**Comment by Feyensv** [comment]
Well, the global change to create multis should be more seen like a guideline for perf and avoiding bad examples remaining in the code, but I'd say it's still not a true necessity (and we probably want to leave some leeway/time to community/internal/... modules for their conversion to the multi API :thinking:).

**Pull request closed without merging**